homeassistant/helpers/script.py

```diff
@@ -1024,6 +1028,27 @@ async def async_run_with_trace(idx: int, script: Script) -> None:
             if isinstance(result, Exception):
                 raise result
 
+    async def _async_data_sources_step(self):
+        """Handle data sources."""
+        self._step_log("data sources")
+        data_sources = self._action[CONF_DATA_SOURCES]
+        tasks = []
+        for _, config in data_sources.items():
+            try:
+                items = template.render_complex(config, self._variables)
+            except (exceptions.TemplateError, ValueError) as ex:
```

Comment:
Do you know how to write a test case that will cause this to fail? When i set an invalid template it is handled by a config validation step. Invalid variables are handled with a warning and don't raise.